### PR TITLE
[WIP] first updates to introduce sockets

### PR DESCRIPTION
**Pull request by ibaryshnikov**
It's an attempt to implement two first networking methods:
- `sock_socket`, which creates a file descriptor and stores some information like domain, type, protocol
- `sock_connect`, which opens connection using fd obtained in `sock_socket`

### CODE REVIEW

File: src/fdentry.rs [hunk 1]
```diff
@@ -2,11 +2,20 @@ use crate::sys::fdentry_impl::{determine_type_and_access_rights, OsFile};
 use crate::{host, Error, Result};
 use std::mem::ManuallyDrop;
 use std::path::PathBuf;
-use std::{fs, io};
+use std::{fs, io, net};
+
+#[derive(Debug)]
+pub struct SocketDetails {
+    pub socket_domain: i32,
+    pub socket_type: u8,
```
Comment (steveej) [inline src/fdentry.rs:10]: @ibaryshnikov have you considered an enum type here instead? What would be the down/upsides to that?

**Comment by marmistrz** [comment]
If you're introducing socket support, you probably should also take care of this issue: https://github.com/CraneStation/wasi-common/issues/167

**Comment by kubkon** [comment]
Hey @ibaryshnikov! Would it be OK if you continued your work directly in `wasmtime` repo after CraneStation/wasmtime#502 lands?

**Comment by ibaryshnikov** [comment]
@kubkon yes, totally fine. I'll need to open a new PR directly in wasmtime, right?

File: src/fdentry.rs (hunk 1 above)
Comment (ibaryshnikov) [reply src/fdentry.rs:10]: I guess it may be useful, I'd like to try. It's a common pattern while working with FFI to provide safe wrappers

**Comment by ibaryshnikov** [comment]
@marmistrz thanks, I'll take a look

**Comment by kubkon** [comment]
> @kubkon yes, totally fine. I'll need to open a new PR directly in wasmtime, right?

Yep, that's the idea :-)

**Comment by kubkon** [comment]
I'm gonna go ahead and close this PR. @ibaryshnikov please reopen it directly in [wasmtime](https://github.com/cranestation/wasmtime).

**Pull request closed without merging**